lib/kubeclient/common.rb

```diff
@@ -114,8 +116,8 @@ def handle_exception
         {}
       end
       err_message = json_error_msg['message'] || e.message
-      error_klass = e.http_code == 404 ? ResourceNotFoundError : HttpError
-      raise error_klass.new(e.http_code, err_message, e.response)
+      error_klass = e.http_code == 404 || e.instance_of?(RestClient::NotFound) ? ResourceNotFoundError : HttpError
```

Comment:
this is getting a little too complicated for a ternary expression. 
Consider switching it to a dumb if